connector/exceptionsconnector/connector_logs.go

```diff
@@ -0,0 +1,154 @@
+func (c *logsConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// ConsumeTraces implements the consumer.Traces interface.
+// It aggregates the trace data to generate logs.
+func (c *logsConnector) ConsumeTraces(ctx context.Context, traces ptrace.Traces) error {
+	ld := plog.NewLogs()
+	for i := 0; i < traces.ResourceSpans().Len(); i++ {
+		rspans := traces.ResourceSpans().At(i)
+		resourceAttr := rspans.Resource().Attributes()
+		serviceAttr, ok := resourceAttr.Get(conventions.AttributeServiceName)
+		if !ok {
+			continue
+		}
+		serviceName := serviceAttr.Str()
+		ilsSlice := rspans.ScopeSpans()
+		for j := 0; j < ilsSlice.Len(); j++ {
+			sl := c.newScopeLogs(ld)
+			ils := ilsSlice.At(j)
+			ils.Scope().CopyTo(sl.Scope())
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				for l := 0; l < span.Events().Len(); l++ {
+					event := span.Events().At(l)
+					if event.Name() == "exception" {
+						c.attrToLogRecord(sl, serviceName, span, event)
+					}
+				}
+			}
+		}
+	}
+	return c.exportLogs(ctx, ld)
+}
+
+func (c *logsConnector) exportLogs(ctx context.Context, ld plog.Logs) error {
+	if err := c.logsConsumer.ConsumeLogs(ctx, ld); err != nil {
+		c.logger.Error("failed to convert exceptions to logs", zap.Error(err))
+		return err
+	}
+	return nil
+}
+
+func (c *logsConnector) newScopeLogs(ld plog.Logs) plog.ScopeLogs {
+	rl := ld.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
+	return sl
+}
+
+func (c *logsConnector) attrToLogRecord(sl plog.ScopeLogs, serviceName string, span ptrace.Span, event ptrace.SpanEvent) plog.LogRecord {
+	logRecord := sl.LogRecords().AppendEmpty()
+
+	logRecord.SetTimestamp(event.Timestamp())
+	logRecord.SetSeverityNumber(plog.SeverityNumberError)
+	logRecord.SetSeverityText("ERROR")
+	eventAttrs := event.Attributes()
+	spanAttrs := span.Attributes()
+
+	// Add common attributes to the log record.
+	logRecord.Attributes().PutStr(spanKindKey, traceutil.SpanKindStr(span.Kind()))
+	logRecord.Attributes().PutStr(statusCodeKey, traceutil.StatusCodeStr(span.Status().Code()))
+	logRecord.Attributes().PutStr(serviceNameKey, serviceName)
+
+	// Add configured dimension attributes to the log record.
+	for _, d := range c.dimensions {
+		if v, ok := getDimensionValue(d, spanAttrs, eventAttrs); ok {
+			logRecord.Attributes().PutStr(d.name, v.Str())
+		}
+	}
+
+	// Add stacktrace to the log record.
+	logRecord.Attributes().PutStr(exceptionStacktraceKey, getValue(eventAttrs, exceptionStacktraceKey))
+
+	// Add HTTP context to the log record.
+	for k, v := range extractHTTP(spanAttrs) {
+		logRecord.Attributes().PutStr(k, v)
+	}
+	return logRecord
+}
+
+// extractHTTP extracts the HTTP context from span attributes.
+func extractHTTP(attr pcommon.Map) map[string]string {
+	http := make(map[string]string)
+	attr.Range(func(k string, v pcommon.Value) bool {
+		if strings.HasPrefix(k, "http.") {
```

Comment:
This answers my previous question. The readme has to tell users to expect all attributes starting with "http." are going to be copied over.